AbstractJanusGraphAssemblyIT.command

## Flaky Test

* **Test Name:** `JanusGraphScriptIT.testGraphOfTheGodsGraphSON:26->AbstractJanusGraphAssemblyIT.unzipAndRunExpect:106->AbstractJanusGraphAssemblyIT.unzipAndRunExpect:81->AbstractJanusGraphAssemblyIT.parseTemplateAndRunExpect:102->AbstractJanusGraphAssemblyIT.expect:121->AbstractJanusGraphAssemblyIT.command:160 expected: <0> but was: <1>`
* **Branch:** `master`

### Stack Trace

```[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 69.586 s <<< FAILURE! - in org.janusgraph.pkgtest.JanusGraphScriptIT

[ERROR] testGraphOfTheGodsGraphSON  Time elapsed: 69.393 s  <<< FAILURE!

org.opentest4j.AssertionFailedError: expected: <0> but was: <1>

	at org.janusgraph.pkgtest.JanusGraphScriptIT.testGraphOfTheGodsGraphSON(JanusGraphScriptIT.java:26)

[INFO] 

[INFO] Results:

[INFO] 

[ERROR] Failures: 

[ERROR]   JanusGraphScriptIT.testGraphOfTheGodsGraphSON:26->AbstractJanusGraphAssemblyIT.unzipAndRunExpect:106->AbstractJanusGraphAssemblyIT.unzipAndRunExpect:81->AbstractJanusGraphAssemblyIT.parseTemplateAndRunExpect:102->AbstractJanusGraphAssemblyIT.expect:121->AbstractJanusGraphAssemblyIT.command:160 expected: <0> but was: <1>

[INFO] 

[ERROR] Tests run: 6, Failures: 1, Errors: 0, Skipped: 0
```

--- a/janusgraph-dist/pom.xml
+++ b/janusgraph-dist/pom.xml
@@ -52,7 +52,7 @@
         <dependency>
             <groupId>io.airlift</groupId>
             <artifactId>airline</artifactId>
-            <version>0.9</version>
+            <version>0.6</version>
         </dependency>
         <dependency>
             <groupId>org.apache.tinkerpop</groupId>

--- a/janusgraph-dist/src/assembly/descriptor/common.component.xml
+++ b/janusgraph-dist/src/assembly/descriptor/common.component.xml
@@ -27,6 +27,7 @@
             <filtered>false</filtered>
             <excludes>
               <exclude>**/elasticsearch.yml</exclude>
+              <exclude>**/jvm.options</exclude>
               <exclude>**/data/**</exclude>
               <exclude>**/logs/**</exclude>
             </excludes>

--- a/janusgraph-dist/src/assembly/static/bin/janusgraph.sh
+++ b/janusgraph-dist/src/assembly/static/bin/janusgraph.sh
@@ -152,7 +152,7 @@ start() {
         CASSANDRA_INCLUDE="$BIN"/cassandra.in.sh "$BIN"/cassandra >/dev/null 2>&1 || exit 1
     fi
     wait_for_cassandra || {
-        echo "See $BIN/../log/cassandra.log for Cassandra log output."    >&2
+        echo "See $BIN/../logs/cassandra.log for Cassandra log output."    >&2
         return 1
     }
 
@@ -164,7 +164,7 @@ start() {
         "$BIN"/../elasticsearch/bin/elasticsearch -d >/dev/null 2>&1
     fi
     wait_for_startup Elasticsearch $ELASTICSEARCH_IP $ELASTICSEARCH_PORT $ELASTICSEARCH_STARTUP_TIMEOUT_S || {
-        echo "See $BIN/../log/elasticsearch.log for Elasticsearch log output."  >&2
+        echo "See $BIN/../logs/elasticsearch.log for Elasticsearch log output."  >&2
         return 1
     }
 
@@ -176,7 +176,7 @@ start() {
         "$BIN"/gremlin-server.sh conf/gremlin-server/gremlin-server.yaml >/dev/null 2>&1 &
     fi
     wait_for_startup 'Gremlin-Server' $GSRV_IP $GSRV_PORT $GSRV_STARTUP_TIMEOUT_S || {
-        echo "See $BIN/../log/gremlin-server.log for Gremlin-Server log output."  >&2
+        echo "See $BIN/../logs/gremlin-server.log for Gremlin-Server log output."  >&2
         return 1
     }
     disown
@@ -239,7 +239,7 @@ clean() {
         echo 'Data directory does not exist.' >&2
     fi
 
-    if cd "$BIN"/../log; then
+    if cd "$BIN"/../logs; then
         rm -f cassandra*.log
         rm -f elasticsearch*.log
         rm -f gremlin-server.log

--- a/janusgraph-dist/src/assembly/static/conf/es/jvm.options
+++ b/janusgraph-dist/src/assembly/static/conf/es/jvm.options
@@ -1,0 +1,91 @@
+# Copyright 2020 JanusGraph Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+## JVM configuration
+
+################################################################
+## IMPORTANT: JVM heap size
+################################################################
+##
+## You should always set the min and max JVM heap
+## size to the same value. For example, to set
+## the heap to 4 GB, set:
+##
+## -Xms4g
+## -Xmx4g
+##
+## See https://www.elastic.co/guide/en/elasticsearch/reference/current/heap-size.html
+## for more information
+##
+################################################################
+
+# Xms represents the initial size of total heap space
+# Xmx represents the maximum size of total heap space
+
+-Xms1g
+-Xmx1g
+
+################################################################
+## Expert settings
+################################################################
+##
+## All settings below this section are considered
+## expert settings. Don't tamper with them unless
+## you understand what you are doing
+##
+################################################################
+
+## GC configuration
+8-13:-XX:+UseConcMarkSweepGC
+8-13:-XX:CMSInitiatingOccupancyFraction=75
+8-13:-XX:+UseCMSInitiatingOccupancyOnly
+
+## G1GC Configuration
+# NOTE: G1 GC is only supported on JDK version 10 or later
+# to use G1GC, uncomment the next two lines and update the version on the
+# following three lines to your version of the JDK
+# 10-13:-XX:-UseConcMarkSweepGC
+# 10-13:-XX:-UseCMSInitiatingOccupancyOnly
+14-:-XX:+UseG1GC
+14-:-XX:G1ReservePercent=25
+14-:-XX:InitiatingHeapOccupancyPercent=30
+
+## JVM temporary directory
+-Djava.io.tmpdir=${ES_TMPDIR}
+
+## heap dumps
+
+# generate a heap dump when an allocation from the Java heap fails
+# heap dumps are created in the working directory of the JVM
+-XX:+HeapDumpOnOutOfMemoryError
+
+# specify an alternative path for heap dumps; ensure the directory exists and
+# has sufficient space
+-XX:HeapDumpPath=data
+
+# specify an alternative path for JVM fatal error logs
+-XX:ErrorFile=../logs/hs_err_pid%p.log
+
+## JDK 8 GC logging
+8:-XX:+PrintGCDetails
+8:-XX:+PrintGCDateStamps
+8:-XX:+PrintTenuringDistribution
+8:-XX:+PrintGCApplicationStoppedTime
+8:-Xloggc:logs/gc.log
+8:-XX:+UseGCLogFileRotation
+8:-XX:NumberOfGCLogFiles=32
+8:-XX:GCLogFileSize=64m
+
+# JDK 9+ GC logging
+9-:-Xlog:gc*,gc+age=trace,safepoint:file=../logs/gc.log:utctime,pid,tags:filecount=32,filesize=64m

